Added -g/--get-hunks command line argument

jamesl33 opened the pull request:
Added a new command line argument -g/--get-hunks which parses all of the hunks in a patch
file as described in #2.

calixteman commented:
@jamesl33, it looks pretty good :)
Could you fix flake8 issues ?
Could you add a test (with several hunks and files) ?

jamesl33 commented:
@calixteman Thanks.
I have added a unit test and fixed any flake8 warnings/errors.
If you would like me to add another unit test or know a way that would mean they
wouldn't need to be hard coded please let me know.

calixteman commented on parsepatch/patch.py:
```diff
@@ -64,11 +65,11 @@ def lines_chunk(it):
         )
         p.changeset = chgset
-        return p.parse()
+        return p.parse(get_hunks=get_hunks)
```
nit: to be coherent, you should put the option in the Patch ctor

jamesl33 replied:
I have addressed this by adding get_hunks to the Patch constructor.

calixteman commented on parsepatch/patch.py:
```diff
@@ -81,14 +82,24 @@ def gen(x):
         with open(filename, 'r') as In:
             patch = In.read()
+
+            # Patch files may contain an empty newline. Remove it.
```
nit: it isn't related to get_hunks stuff, so make an other PR for that stuff

jamesl33 replied:
I will open another pull request for this but It will need to be pulled before #3 otherwise the testing will fail.

calixteman commented on parsepatch/patch.py:
```diff
@@ -286,6 +297,34 @@ def parse_hunks(self, line):
+        diff = []
+
+        for h_line in self.get_lines:
+            if h_line.startswith('diff --git a/') or h_line.startswith('diff -r '):
```
Is there a way to avoid code duplication ?
I mean we've common code here and in get_hunks(self), so it'd be nice to avoid it.

jamesl33 replied:
I have now addressed the code duplication by merging get_changes(self) and get_hunks(self).

calixteman commented:
@La0 does this patch fulfill your expectations ?

jamesl33 commented:
I've updated the commits in response to @calixteman. The tests will fail until #4 is accepted because this commit makes use of the skip_newlines(self) function.

jamesl33 commented:
Hi, it appears that you may have beaten me too it, I have rebased this branch on the master branch with the changes you made in commit 9f79d95. If the changes are already merged please feel free to close this pull request. Thanks for the help you provided me during the time I worked on issue #2.
